server/src/test/java/org/opensearch/action/admin/cluster/node/tasks/ResourceAwareTasksTests.java

```diff
@@ -616,6 +617,10 @@ public void onFailure(Exception e) {
 
         taskTestContext.requestCompleteLatch.await();
 
+        // It is possible for the MockTaskManagerListener to be called after the response is sent already.
+        // Wait enough time for taskId to be added to taskIdsRemovedFromThreadContext before performing validations.
+        waitUntil(() -> taskIdsAddedToThreadContext.size() == taskIdsRemovedFromThreadContext.size(), 5, TimeUnit.SECONDS);
+
```

Comment:
I am assuming this timeout should be good enough and we have tested multiple test iterations